cli/src/pcluster/models/cluster_resources.py

```diff
@@ -172,15 +179,33 @@ class ClusterCreationFailure:
                 "Cluster has been set to PROTECTED mode due to failures detected in static node provisioning",
             ),
         ]
+        general_failure = ClusterCreationFailure("ClusterCreationFailure", "Failed to create the cluster.", "")
+        return cluster_creation_failures, general_failure
+
+    def _update_failures(self):
+        @dataclass
+        class ClusterUpdateFailure:
+            """Represent the object holding the data of ClusterCreationFailure."""
+
+            failure_code: str
+            api_failure_reason: str
+            cfn_failure_reason: str  # cloud formation matching string
+
+        cluster_update_failures = [
+            ClusterUpdateFailure("EbsMountFailure", "Failed to mount EBS volume.", "Failed to mount EBS volume"),
+            ClusterUpdateFailure("EfsMountFailure", "Failed to mount EFS.", "Failed to mount EFS"),
+            ClusterUpdateFailure("FsxMountFailure", "Failed to mount FSX.", "Failed to mount FSX"),
+            ClusterUpdateFailure("RaidMountFailure", "Failed to mount RAID array.", "Failed to mount RAID array"),
+        ]
+        general_failure = ClusterUpdateFailure("ClusterUpdateFailure", "Failed to update the cluster.", "")
+        return cluster_update_failures, general_failure
```

Comment:
Why do we change the return here?

Reply:
There are two type of failures and each has its own general failure message.